### ISSUE TITLE
the negative_weights seems no used in rpn-box-loss

### ISSUE THREAD

**Issue by z-huabao** [jwyang/faster-rcnn.pytorch#635]
Hello, I am learning the code of faster-rcnn, but I think it may be a mistake here:

[anchor_target_layer.py#L164](https://github.com/jwyang/faster-rcnn.pytorch/tree/master/lib/model/rpn/anchor_target_layer.py#L164)
```
        bbox_inside_weights = gt_boxes.new(batch_size, inds_inside.size(0)).zero_()  # [n_batch, n_anchor]
        ...
        bbox_inside_weights[labels==1] = cfg.TRAIN.RPN_BBOX_INSIDE_WEIGHTS[0]

        if cfg.TRAIN.RPN_POSITIVE_WEIGHT < 0:
            num_examples = torch.sum(labels[i] >= 0)
            positive_weights = 1.0 / num_examples.item()  # will use to compute the mean loss along all matched anchor-boxes 
            negative_weights = 1.0 / num_examples.item()
        else:
            assert ((cfg.TRAIN.RPN_POSITIVE_WEIGHT > 0) &
                    (cfg.TRAIN.RPN_POSITIVE_WEIGHT < 1))

        bbox_outside_weights[labels == 1] = positive_weights
        bbox_outside_weights[labels == 0] = negative_weights  # <-- it seems no used???
```

`negative_weights` above(last line) seems no used, because the `in_loss_box` bellow which has multiply by `bbox_inside_weights`, where `[labels == 0]` values should all be zero.

so `in_box_diff`, `in_loss_box`, `bbox_outside_weights * in_loss_box` bellow where `[labels == 0]` all zeros too, so I think `negative_weights` above can be ignored.

[net_utils.py#L73](https://github.com/jwyang/faster-rcnn.pytorch/tree/master/lib/model/utils/net_utils.py#L73)
```
def _smooth_l1_loss(bbox_pred, bbox_targets, bbox_inside_weights, bbox_outside_weights, sigma=1.0, dim=[1]):

    sigma_2 = sigma ** 2
    box_diff = bbox_pred - bbox_targets
    in_box_diff = bbox_inside_weights * box_diff  # <------------------------
    abs_in_box_diff = torch.abs(in_box_diff)
    smoothL1_sign = (abs_in_box_diff < 1. / sigma_2).detach().float()
    in_loss_box = torch.pow(in_box_diff, 2) * (sigma_2 / 2.) * smoothL1_sign \
                  + (abs_in_box_diff - (0.5 / sigma_2)) * (1. - smoothL1_sign)
    out_loss_box = bbox_outside_weights * in_loss_box  # <--------------------------
```

if I am right, `num_examples = torch.sum(labels[i] >= 0)` should modify to `num_examples = torch.sum(labels[i] > 0)`.

and refrence [issue 256](https://github.com/jwyang/faster-rcnn.pytorch/issues/256), in my idea, 
`num_examples = torch.sum(labels > 0, axis=1).mean()`

other correlate issue: [issue 558](https://github.com/jwyang/faster-rcnn.pytorch/issues/558)
